spec/intl.html

```diff
@@ -128,7 +128,11 @@ <h1>Intl.supportedValuesOf ( _key_ )</h1>
       <emu-alg>
         1. Let _key_ be ? ToString(_key_).
         1. If _key_ is *"calendar"*, then
-          1. Let _list_ be AvailableCanonicalCalendars( ).
+          1. Let _list_ be a new empty List.
+          1. For each element _identifier_ of AvailableCalendars(), do
+            1. Let _canonical_ be CanonicalizeCalendar(_identifier_).
+            1. If _identifier_ is _canonical_, then
+              1. Append _canonical_ to _list_.
```

Comment:
Rather than introducing a calendar-specific operation, I'd like one that covers all "-u-…" keywords and is also used in [ResolveLocale](https://tc39.es/ecma402/#sec-resolvelocale) step 12.g.iii.
```suggestion
            1. Let _canonical_ be CanonicalizeUValue(*"ca"*, _identifier_).
            1. If _identifier_ is _canonical_, then
              1. Append _identifier_ to _list_.
```

```
CanonicalizeUValue (
  _ukey_: a Unicode locale extension sequence key defined in <a href="https://unicode.org/reports/tr35/#Key_And_Type_Definitions_">Unicode Technical Standard #35 Part 1 Core, Section 3.6.1 Key and Type Definitions</a>,
  _uvalue_: a String,
): a String
```
```
1. Let _lowerValue_ be the ASCII-lowercase of _uvalue_.
1. Let _canonicalized_ be the String value resulting from canonicalizing _lowerValue_ as a value of key _ukey_ per <a href="https://unicode.org/reports/tr35/#processing-localeids">Unicode Technical Standard #35 Part 1 Core, Annex C LocaleId Canonicalization Section 5 Canonicalizing Syntax, Processing LocaleIds</a>.
1. NOTE: It is recommended that implementations use the 'u' extension data in <code>common/bcp47</code> provided by the Common Locale Data Repository (available at <a href="https://cldr.unicode.org/">https://cldr.unicode.org/</a>).
1. Return _canonicalized_.
```